### ISSUE TITLE
TST & TSM:  Loading saved session results in different results (FF57b12)

### ISSUE THREAD

**Issue by irvinm** [sienori/Tab-Session-Manager#20]
1. Start empty browser with only TST and TSM as add-ons.
2. Load "Save" stored session.  (Now have 56 tabs in a new window)
3. Load "Save" stored session.  (Now have 56 tabs in a new window)
4. Load "Save" stored session.  (Now have 56 tabs in a new window)
5. Load "Save" stored session.  (Now have 56 tabs in a new window)
6. Collapse all "trees" at their root to just see high level tab structure
7. Notice that the resulting tabs are different in the 4 new windows

![2017-10-28_21-09-00](https://user-images.githubusercontent.com/979729/32139878-0a653f36-bc25-11e7-97f7-e35d478ddd75.png)

I have attached the storage.js which contains the stored session.

[storage.zip](https://github.com/sienori/Tab-Session-Manager/files/1424732/storage.zip)

Looking more closely at the 3rd and 4th attempt differences, you can see that the tree with "Browser Extensions - ... (3)" and tree "firefox css generator - ... (4)" both ended up in the GitHub tree at the bottom.

![2017-10-28_21-27-34a](https://user-images.githubusercontent.com/979729/32143249-fa0ab0be-bc7c-11e7-8fb9-bf073195353a.png)

This could also be a TST issue or even a FF race condition or something, but seeing as I could reproduce it TSM I thought I would pass it along for any feedback.

**Comment by piroor** [comment]
By the commit https://github.com/piroor/treestyletab/commit/7230f32 TST now updates `openerTabId` with valid value, so I hope that updated `openerTabId` information is saved by TSM and tree is restored as expected.

**Comment by irvinm** [comment]
I retested this issue today and got the same results.  (Varying restore tab groupings using the same session)

However, while doing this I got to thinking about what might be causing this.  I realized there is an option in TST to auto-group tabs if they are opened within a certain time frame.

**I disabled this option and reran the test several times and I get consistent results now!**  (This makes sense as the session is restored tab-by-tab and if the system were able to open a certain tab within 100ms of the previous one, TST would group them together thus giving the appearance of random issues and\or parent-child issues.)

Thank you @piroor and @sienori for working on these great add-ons.

Sincerely,
Mike
